connector/connect/server/src/main/scala/org/apache/spark/sql/connect/service/SparkConnectService.scala

```diff
@@ -259,13 +281,15 @@ object SparkConnectService {
     if (debugMode) {
       sb.addService(ProtoReflectionService.newInstance())
     }
+
     server = sb.build
     server.start()
+    createListenerAndUI(sparkContextOpt.getOrElse(SparkSession.active.sparkContext))
   }
 
   // Starts the service
-  def start(): Unit = {
-    startGRPCService()
+  def start(sparkContextOpt: Option[SparkContext] = None): Unit = {
```

Comment:
make it `sparkContext: SparkContext` , not an option, and in SparkConnectServer pass it explcitly from the session there. better that than having to `.getOrElse(SparkSession.active.sparkContext)` without controlling if there's an active session here